### ISSUE TITLE
Failure to boot on linux

### ISSUE THREAD

**Issue by LuxLucian** [emoisback/Helldivers-2-Internal-Hack-Dll-Proxy-PoC#47]
I've finally got the DLL compiling on my laptop, but it still crashes when I try to use it on Linux(EndeavorOS). If there is any way to find out what is going wrong it would be appreciated. I have a guess that the console pop up might have something to do with it, maybe the config version might work.

**Comment by LuxLucian** [comment]
So far I have tried with multiple DLLs, the last two attempts being the latest git compiled. I also tried gutting the script to just include the stratagem unlocks and it did not work.

**Comment by Atamg1994** [comment]
This library will not work on Linux

the library essentially redefines the functions of the Windows standard library and injects code in one go

**Comment by LuxLucian** [comment]
Thanks for the info, are there other ways to inject code that might work on linux that I can look into?

**Comment by baesik** [comment]
> Thanks for the info, are there other ways to inject code that might work on linux that I can look into?

From the very limited amount of research into the linux version, the product they use is disabled for Linux compatibility and, assuming, memory scanners work like normal on the game.

**Comment by Atamg1994** [comment]
try to study how the library works in this repository, you should find something similar for Linux
https://github.com/NeighTools/UnityDoorstop.Unix
but this one loads the c# unity engine

and you need to do similar proxying but load C++ code

**Comment by Kahdeg-15520487** [comment]
if you are using steam proton, i have managed to inject it
add this to steam launch options
`WINEDLLOVERRIDES="version.dll=n,b" %command%`
proof (i inject my conf version so it create the conf file):
![image](https://github.com/emoisback/Helldivers-2-Internal-Hack-Dll-Proxy-PoC/assets/15503419/f28def88-1d0c-41f7-aad0-48ffc38b7211)

**Comment by LuxLucian** [comment]
Were you able to start the game with it? I was using WINEDLLOVERRIDES but the game would not start up when I did it.

**Comment by Kahdeg-15520487** [comment]
yes, i was using basic debian 12 install with steam proton experimental

**Comment by LuxLucian** [comment]
Using the same launch args and a compiled version from my laptop has the same result, steam just goes back to PLAY and nothing seems to happen. No conf file is generated either.

**Comment by LuxLucian** [comment]
> yes, i was using basic debian 12 install with steam proton experimental

Does running the launch arg with no version.dll present launch the game normally for you?

**Comment by Kahdeg-15520487** [comment]
yes, it can launch normally

**Comment by LuxLucian** [comment]
Don't know what's causing steam to not launch the game then. I'll keep poking around and see if something comes up. I assume you ran the game on a normal prefix with nothing extra installed on it?

**Comment by LuxLucian** [comment]
Using PROTON_LOG=1 doesn't even create a log, I think this is a steam issue.

**Comment by DeathWrench** [comment]
Someone posted this: https://www.unknowncheats.me/forum/4013093-post465.html
Don't know if it will work for you.

**Comment by gir489returns** [comment]
On Linux, GameGuard can't run its rootkit, so it just gives up and doesn't do anything. Would probably be easier for Linux users to just use protonhax and run Cheat Engine inside of the Proton instance.

**Comment by LuxLucian** [comment]
Finally got the game to make a proton log, looks like it's not finding DLLs.
[steam-553850.log](https://github.com/emoisback/Helldivers-2-Internal-Hack-Dll-Proxy-PoC/files/14533054/steam-553850.log)

**Comment by LuxLucian** [comment]
Looking in the prefix they are there, it just seems like they can't be detected.

**Comment by LuxLucian** [comment]
> On Linux, GameGuard can't run its rootkit, so it just gives up and doesn't do anything. Would probably be easier for Linux users to just use protonhax and run Cheat Engine inside of the Proton instance.

I'm unfortunately using this for the stratagem/weapon unlocker, and I did't see those cheats in the cheat table when I checked.

**Comment by emoisback** [comment]
Its shared on fearless, as far as i know but not in form of CT

**Comment by LuxLucian** [comment]
Would it be possible to make a CT that does that? I'll start poking around for now.

**Comment by emoisback** [comment]
Yes it is that not hard because you only need chang 3 first byte of function.

**Comment by LuxLucian** [comment]
> Yes it is that not hard because you only need chang 3 first byte of function.

Could you help me figure out what each number he sent is for?
Patch game.dll+0xB7B690 (0x180B7B690) to B0 01 C3
I don't really know what either of the first two numbers are for. I also don't see any patterns where I would insert the new data. I can't figure out how your code does it either.

**Comment by LuxLucian** [comment]
I got it sorted out by nabbing gir489's table and copying some of the code from your DLL. Thanks for the help! I'll post it in the FR thread.
